pipeline/flows/notifications.py

```diff
@@ -0,0 +1,144 @@
+    }
+    fake_extracted_chats = [
+        ChatData(
+            release_timestamp=datetime.datetime.now(),
+            participants=participants,
+            messages=messages,
+        ),
+        ChatData(release_timestamp=datetime.datetime.now()),
+        ChatData(release_timestamp=datetime.datetime.now()),
+    ]
+    return fake_extracted_chats
+
+
+@prefect.task
+def get_notifications(inp: List[ChatData]) -> List[NotificationInfo]:
+    logger = prefect.context.get("logger")
+    # add other types of notifiers to list once they are implemented
+    logger.info("Generating notifications")
+    notif_info_lst = []
+    for chatdata in inp:
+        notif_input = NotificationInput(
+            chatdata=chatdata, run_datetime=datetime.datetime.now()
+        )
+        for notifier in NOTIFIERS:
+            notif_info = notifier.get_notifications(inp=notif_input)
+            notif_info_lst.append(notif_info)
+    # flatten notif_info_lst, which is a list of lists
+    flat_notif_info_lst = list(itertools.chain(*notif_info_lst))
+    logger.info(
+        f"Done. Number of notifications generated: {len(flat_notif_info_lst)}"
+    )
+    return flat_notif_info_lst
+
+
+@prefect.task
+def notif_hierarchy(notifs: List[NotificationInfo]) -> List[NotificationInfo]:
+    """
+    Truncates the NotificationInfo list based on a hierarchy of notifications, which is as follows:
+    1. check match (highest)
+    2. check message
+    3. rate match
+    4. edit profile (lowest)
+    """
+    truncated_notif_info = {}
+    for notif in notifs:
+        if notif.notification_type.value == "check_match":
+            truncated_notif_info[notif.recipient.uid] = notif
+        elif notif.notification_type.value == "check_message":
+            notif_in_trunc = truncated_notif_info.get(notif.recipient.uid)
+            if notif_in_trunc is None:
+                truncated_notif_info[notif.recipient.uid] = notif
+            elif notif_in_trunc.notification_type.value != "check_match":
+                truncated_notif_info[notif.recipient.uid] = notif
+        elif notif.notification_type.value == "rate_match":
+            notif_in_trunc = truncated_notif_info.get(notif.recipient.uid)
+            if notif_in_trunc is None:
+                truncated_notif_info[notif.recipient.uid] = notif
+            elif notif_in_trunc.notification_type.value == "edit_profile":
+                truncated_notif_info[notif.recipient.uid] = notif
+        else:
+            truncated_notif_info[notif.recipient.uid] = notif
+    return list(truncated_notif_info.values())
+
+
+@prefect.task
+def render_notifications(
+    notifs: List[NotificationInfo],
+) -> List[str]:
+    logger = prefect.context.get("logger")
+    output_lst = []
+    for notif in notifs:
+        renderer = RENDERERS.get(notif.notification_type)
+        if renderer:
+            output_str = renderer.render(notif)
+            logger.info(output_str)
```

Comment:
In the future, we need to add a `NotificationOutput` dataclass that includes the recipient email, subject, and body.